### ISSUE TITLE
Productos repetidos en listas.

### ISSUE THREAD

**Issue by luismvargasg** [colav/impactu#104]
Realizando la consulta de la lista de productos de la UdeA se evidencia que hay productos repetidos que se entregan en varias páginas distintas.
Aparentemente se están entregando 3575 artículos repetidos, que pueden repetirse 1 o más veces por lo tanto el número de artículos únicos puede ser inferior:
![image](https://github.com/colav/impactu/assets/60363752/2c2a77ac-d3a4-491b-b315-c5423cf85330)
Ej.1:
**Página 37:** https://api.dev.impactu.colav.co/affiliation/institution/665e0a57aa7c2077adf4e441/research/products?max=250&page=37&sort=citations-
**Página 38:** https://api.dev.impactu.colav.co/affiliation/institution/665e0a57aa7c2077adf4e441/research/products?max=250&page=38&sort=citations-
**Artículo repetido:** "Sueros antiofídicos en Colombia: análisis de la producción, abastecimiento y recomendaciones para el mejoramiento de la red de producción"
**ID a verificar:** 66623634c035cef7eca6fcbb 

Ej.2:
**Página 54:** https://api.dev.impactu.colav.co/affiliation/institution/665e0a57aa7c2077adf4e441/research/products?max=250&page=54&sort=citations-
**Página 55:** https://api.dev.impactu.colav.co/affiliation/institution/665e0a57aa7c2077adf4e441/research/products?max=250&page=55&sort=citations-
**Artículo repetido:** "Synthesis, In Vitro Antiprotozoal Activity and Cytotoxicity of New Thymol Carbonate Derivatives"
**ID a verificar:** 66623750c035cef7eca726ee

La lista completa de ids duplicados se encuentra acá:
https://drive.google.com/file/d/1gvB2uVSAmcvac0LmPCIrwbFNqE2uXVEE/view?usp=drive_link

**Comment by Simon3640** [comment]
En la misma página no se entregan los ids repetidos, parece que debo revisar mejor el pipeline, porque de forma aislada funciona bien.

**Comment by Simon3640** [comment]
Por otro lado el total de artículos no tiene que ver con esto, ya que base de datos no están repetidos, es un error del skip y el limit

**Comment by Simon3640** [comment]
Investigando un poco más me doy cuenta de que el problema es la combinación entre sort, skip y limit, es simple tenemos artículos con conteo de citaciones iguales, lo cual crea ambiguedad a la hora de paginar.

**Comment by Simon3640** [comment]
@omazapa Se te ocurre algo para solucionar esto?

**Comment by luismvargasg** [comment]
> Investigando un poco más me doy cuenta de que el problema es la combinación entre sort, skip y limit, es simple tenemos artículos con conteo de citaciones iguales, lo cual crea ambiguedad a la hora de paginar.

Efectivamente, haciendo el sort alphabetical da menos errores, solo falla por allá cuando llega a un montón de artículos que se llaman "Editorial" entrega 10 repetidos nada más.
![image](https://github.com/colav/impactu/assets/60363752/756ff8ce-8381-4ab8-881c-5df3d011cdcc)

**Comment by Simon3640** [comment]
https://github.com/colav-playground/impactu-backend-service/commit/9efbca2257d94d9589490117f4d138a4e5590dc8
Corregido

**Comment by luismvargasg** [comment]
Error relacionado con este issue detectado nuevamente en la respuesta de la API.
